modules/auxiliary/gather/memcached_extractor.rb

```diff
@@ -54,12 +54,13 @@ def localhost?(ip)
   def enumerate_keys
     keys = []
     enumerate_slab_ids.each do |sid|
+      sock.send("stats cachedump #{sid} #{max_keys}\r\n", 0)
```

Comment:
Does `max_keys` here end up a reason for iteration or is the response guaranteed to be all keys?

Reply:
So according to https://elijaa.org/2010/12/24/understanding-memcached-stats-cachedump-command.html this is simply the maximum number of keys to dump however supposedly there may be a maximum return size of 1 MB, though given this command isn't documented and is part of the undocumented interface, there is speculation it may be 2 MB but not confirmed.

This is also noted at https://lzone.de/blog/How-to-Dump-Keys-from-Memcache where they note the following constraints:

1. You can only dump keys per slab class (keys with roughly the same content size)
2. You can only dump one page per slab class (1MB of data)
3. This is an unofficial feature that [might be removed anytime.](https://groups.google.com/forum/?fromgroups=#!topic/memcached/1-T8I-RVGKM)

Hopefully that helps explain things a bit here.